### ISSUE TITLE
Glusterfs: could not open log file for pod: pod_name

### ISSUE THREAD

**Issue by SweetOps** [openshift/openshift-ansible#5197]
I can create pvc and can't mount it to pod. When I create pod with pvc I get error:

```sh
MountVolume.SetUp failed for volume "kubernetes.io/glusterfs/f297b47a-88c9-11e7-9976-0ab88d9eb221-pvc-d2362fbb-88c9-11e7-a6ad-0a862c7c70b3" (spec.Name: "pvc-d2362fbb-88c9-11e7-a6ad-0a862c7c70b3") pod "f297b47a-88c9-11e7-9976-0ab88d9eb221" (UID: "f297b47a-88c9-11e7-9976-0ab88d9eb221") with: glusterfs: mount failed: exit status 1 the following error information was pulled from the glusterfs log to help diagnose this issue: glusterfs: could not open log file for pod: gitlab-106655731-ndlm4
```
 
I have tried: 
* disable selinux
* run on all hosts `setsebool virt_use_fusefs on`
* setup with `openshift_docker_selinux_enabled=False` and disable selinux

but without luck.

And of course I've checked:
* glusterfs-client installed on all hosts
* /var/log/glusterfs is writable  (logs without errors)
* firewall rules

OS: 
```sh
CentOS 7
CentOS Atomic
``` 

```sh
ansible --version
ansible 2.3.1.0
  config file = /usr/share/ansible/openshift-ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```

branch: `3.6`

invemtory:
```sh
[OSEv3:children]
masters
nodes
etcd
glusterfs

[OSEv3:vars]
ansible_ssh_user=centos
ansible_become=true
ansible_ssh_transfer_method=scp
deployment_type=origin
containerized=true
os_sdn_network_plugin_name='redhat/openshift-ovs-subnet'
openshift_master_cluster_method=native
openshift_master_cluster_password=admin
openshift_release=3.6
openshift_image_tag=v3.6.0
openshift_hosted_metrics_deployer_version=v3.6.0
openshift_router_selector='region=infra'
openshift_registry_selector='region=infra'
openshift_schedulable=true
openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
openshift_install_examples=true
### Monitoring
# Enable cockpit
osm_use_cockpit=true

# Set cockpit plugins
osm_cockpit_plugins=['cockpit-kubernetes']

openshift_hosted_logging_deploy=true
openshift_hosted_metrics_deploy=true


### Disable checks
openshift_disable_check=disk_availability, docker_storage, memory_availability


### Enable basic auth
# Local path to your file for basic auth
openshift_master_htpasswd_file='/tmp/auth'
###
openshift_master_identity_providers=[ {'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'} ]
openshift_master_default_subdomain=domain.com
openshift_master_cluster_hostname=domain.com
openshift_master_cluster_public_hostname=domain.com

[etcd]
sir-3y5ihzan
sir-q178j6am
sir-rsbgjr8p

[masters]
sir-3y5ihzan 
sir-q178j6am
sir-rsbgjr8p

[nodes]
sir-3y5ihzan 
sir-q178j6am
sir-rsbgjr8p

[glusterfs]
sir-3y5ihzan  glusterfs_devices='[ "/dev/xvdb" ]'
sir-q178j6am glusterfs_devices='[ "/dev/xvdb" ]'
sir-rsbgjr8p glusterfs_devices='[ "/dev/xvdb" ]
```

**Comment by jarrpa** [comment]
What is `gitlab-106655731-ndlm4`?

**Comment by SweetOps** [comment]
It is name of pod

**Comment by jarrpa** [comment]
Does it show up anywhere in `/var/log/glusterfs` ont he node that tried running the pod?

**Comment by SweetOps** [comment]
> Does it show up anywhere in /var/log/glusterfs ont he node that tried running the pod?

No, it doesn't. I see some records only when create the pvc.

**Comment by jarrpa** [comment]
Can you manually mount the gluster volume on the node that tried to run the pod?

**Comment by SweetOps** [comment]
> Can you manually mount the gluster volume on the node that tried to run the pod?

Should I run it inside gluster pod or on the node?

**Comment by jarrpa** [comment]
No, on the node.

**Comment by SweetOps** [comment]
I have mounted on all nodes:

```sh
sudo mount -t glusterfs 10.0.1.150:/vol_08727838285bee2e1dbc42c4e9273d8b /srv
sudo mount|grep vol_08727838285bee2e1dbc42c4e9273d8b 
10.0.1.150:/vol_08727838285bee2e1dbc42c4e9273d8b on /var/srv type fuse.glusterfs (rw,relatime,user_id=0,group_id=0,default_permissions,allow_other,max_read=131072)
```
then tried to redeploy the pod and got same error:

```sh
  17s		1s		6	kubelet, ip-10-0-1-150.eu-central-1.compute.internal			Warning		FailedMount	MountVolume.SetUp failed for volume "kubernetes.io/glusterfs/1c3bf4ca-88e2-11e7-8c30-0690a9f31663-pvc-8ca4e00b-88df-11e7-acff-0642a7031c1b" (spec.Name: "pvc-8ca4e00b-88df-11e7-acff-0642a7031c1b") pod "1c3bf4ca-88e2-11e7-8c30-0690a9f31663" (UID: "1c3bf4ca-88e2-11e7-8c30-0690a9f31663") with: glusterfs: mount failed: exit status 1 the following error information was pulled from the glusterfs log to help diagnose this issue: glusterfs: could not open log file for pod: jenkins-4-vklrt
```

**Comment by jarrpa** [comment]
Hmm... still uncertain. Pulling in some extra help.

@humblec PTAL?

**Comment by SweetOps** [comment]
I've tried to update api server and same result...

```sh
oc version
oc v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://zaebalgluster.skillbyte.com:8443
openshift v3.7.0-alpha.0+e5b4a1c-454
kubernetes v1.7.0+695f48a16f
```
@jarrpa, could you say on which openshift version and OS you've luck with glusterfs?

**Comment by jarrpa** [comment]
OCP 3.6 and CentOS/RHEL 7.3

**Comment by SweetOps** [comment]
@jarrpa, containerized?

**Comment by jarrpa** [comment]
Containerized GlusterFS + heketi, yes.

**Comment by jorgefromhell** [comment]
Hi @SweetOps, @jarrpa, that is the exact same problem I'm having here .. https://github.com/openshift/openshift-ansible/issues/5118

**Comment by jarrpa** [comment]
@SweetOps @jorgefromhell Are the apps allowed to mount PVs of type glusterfs? E.g. are they part of an SCC that allows:
```
volumes:
- glusterfs
```

**Comment by jorgefromhell** [comment]
@jarrpa, it doesn't, but I checked the other cluster that I have which is working (1.5.1) and I haven't found any SCC with glusterfs in it.

1.5.1:
`[root@app-dc-hom-osmaster01 ~]# oc get scc
NAME               PRIV      CAPS      SELINUX     RUNASUSER          FSGROUP     SUPGROUP    PRIORITY   READONLYROOTFS   VOLUMES
anyuid             false     []        MustRunAs   RunAsAny           RunAsAny    RunAsAny    10         false            [configMap downwardAPI emptyDir persistentVolumeClaim secret]
hostaccess         false     []        MustRunAs   MustRunAsRange     MustRunAs   RunAsAny    <none>     false            [configMap downwardAPI emptyDir hostPath persistentVolumeClaim secret]
hostmount-anyuid   false     []        MustRunAs   RunAsAny           RunAsAny    RunAsAny    <none>     false            [configMap downwardAPI emptyDir hostPath nfs persistentVolumeClaim secret]
hostnetwork        false     []        MustRunAs   MustRunAsRange     MustRunAs   MustRunAs   <none>     false            [configMap downwardAPI emptyDir persistentVolumeClaim secret]
nonroot            false     []        MustRunAs   MustRunAsNonRoot   RunAsAny    RunAsAny    <none>     false            [configMap downwardAPI emptyDir persistentVolumeClaim secret]
privileged         true      []        RunAsAny    RunAsAny           RunAsAny    RunAsAny    <none>     false            [*]
restricted         false     []        MustRunAs   MustRunAsRange     MustRunAs   RunAsAny    <none>     false            [configMap downwardAPI emptyDir persistentVolumeClaim secret]
`

3.6.0:
`[root@app-dc-prd-osmaster01 ~]# oc get scc
NAME               PRIV      CAPS      SELINUX     RUNASUSER          FSGROUP     SUPGROUP    PRIORITY   READONLYROOTFS   VOLUMES
anyuid             false     []        MustRunAs   RunAsAny           RunAsAny    RunAsAny    10         false            [configMap downwardAPI emptyDir persistentVolumeClaim projected secret]
hostaccess         false     []        MustRunAs   MustRunAsRange     MustRunAs   RunAsAny    <none>     false            [configMap downwardAPI emptyDir hostPath persistentVolumeClaim projected secret]
hostmount-anyuid   false     []        MustRunAs   RunAsAny           RunAsAny    RunAsAny    <none>     false            [configMap downwardAPI emptyDir hostPath nfs persistentVolumeClaim projected secret]
hostnetwork        false     []        MustRunAs   MustRunAsRange     MustRunAs   MustRunAs   <none>     false            [configMap downwardAPI emptyDir persistentVolumeClaim projected secret]
nonroot            false     []        MustRunAs   MustRunAsNonRoot   RunAsAny    RunAsAny    <none>     false            [configMap downwardAPI emptyDir persistentVolumeClaim projected secret]
privileged         true      [*]       RunAsAny    RunAsAny           RunAsAny    RunAsAny    <none>     false            [*]
restricted         false     []        MustRunAs   MustRunAsRange     MustRunAs   RunAsAny    <none>     false            [configMap downwardAPI emptyDir persistentVolumeClaim projected secret] 
`

**Comment by jarrpa** [comment]
@jorgefromhell Right, that wouldn't matter... nevermind. :(

**Comment by jorgefromhell** [comment]
@jarrpa reading through the openshift/origin issues, https://github.com/openshift/origin/issues/15950 caught my attention. Do you think it might be related?

**Comment by jarrpa** [comment]
@jorgefromhell Hmm... not impossible. Though I'm not sure why we haven't run into this before.

**Comment by besser82** [comment]
@jarrpa Seeing the same problem over here on fresh install of v3.6 + GlusterFS + Heketi, all containerized on Fedora 26 Atomic.

**Comment by jarrpa** [comment]
Okay, so this seems to be an issue on Atomic. To be fair, I have not had any chance to test the openshift-ansible playbooks on Atomic. I'll need to figure out a way to set up an environment quickly on my test system.

@jorgefromhell @besser82 Could you verify that the `glusterfs-fuse` package is installed? Could someone also quickly replicate the problem and then inspect the logs in `/var/log/glusterfs` on the OCP nodes running the GlusterFS pods to see if there's anything useful?

@SweetOps, I know you've already verified this, my apologies I don't have any new ideas yet. :(

**Comment by SweetOps** [comment]
@jarrpa, looks like issue in iptables rules - I've luck only with enabled  firewalld on Centos/RHEL. Sorry, I hadn't time for fully investigation...

**Comment by jarrpa** [comment]
@SweetOps Ah! Good to know. :)

**Comment by SweetOps** [comment]
@jarrpa, but still need resolve issue with Atomic :(

**Comment by jarrpa** [comment]
@SweetOps If you just had firewall issues, the others might still be missing `mount.glusterfs`. At least these are more tangible problems for me. ;)

**Comment by SweetOps** [comment]
@jarrpa, that's just a guess  :)

**Comment by jorgefromhell** [comment]
@jarrpa I'm not using atomic here ..
`[root@app-dc-prd-osmaster01 ~]# uname -a
Linux app-dc-prd-osmaster01.XXX 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
[root@app-dc-prd-osmaster01 ~]# cat /etc/centos-release
CentOS Linux release 7.3.1611 (Core) 
`
I do have glusterfs-fuse and firewalld installed on all nodes ..
`[jorge.junior@dot197nb Openshift-YAMLS]$ ansible -m raw -i ansible-hosts-openshift-prd -a "yum -y install firewalld glusterfs-fuse" all
app-dc-prd-osetcd01.XXX | SUCCESS | rc=0 >>
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.xpg.com.br
 * epel: mirror.espoch.edu.ec
 * extras: centos.xpg.com.br
 * updates: centos.xpg.com.br
Package firewalld-0.4.3.2-8.1.el7_3.3.noarch already installed and latest version
Package glusterfs-fuse-3.7.9-12.el7.centos.x86_64 already installed and latest version
Nothing to do
Shared connection to app-dc-prd-osetcd01.XXX closed.


app-dc-prd-osnode03.XXX | SUCCESS | rc=0 >>
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.xpg.com.br
 * epel: mirror.espoch.edu.ec
 * extras: centos.xpg.com.br
 * updates: centos.xpg.com.br
Package firewalld-0.4.3.2-8.1.el7_3.3.noarch already installed and latest version
Package glusterfs-fuse-3.7.9-12.el7.centos.x86_64 already installed and latest version
Nothing to do
Shared connection to app-dc-prd-osnode03.XXX closed.


app-dc-prd-osnode01.XXX | SUCCESS | rc=0 >>
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.xpg.com.br
 * epel: mirror.espoch.edu.ec
 * extras: centos.xpg.com.br
 * updates: centos.xpg.com.br
Package firewalld-0.4.3.2-8.1.el7_3.3.noarch already installed and latest version
Package glusterfs-fuse-3.7.9-12.el7.centos.x86_64 already installed and latest version
Nothing to do
Shared connection to app-dc-prd-osnode01.XXX closed.


app-dc-prd-osnode02.XXX | SUCCESS | rc=0 >>
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: ftp.unicamp.br
 * epel: mirror.espoch.edu.ec
 * extras: ftp.unicamp.br
 * updates: ftp.unicamp.br
Package firewalld-0.4.3.2-8.1.el7_3.3.noarch already installed and latest version
Package glusterfs-fuse-3.7.9-12.el7.centos.x86_64 already installed and latest version
Nothing to do
Shared connection to app-dc-prd-osnode02.XXX closed.


app-dc-prd-osetcd03.XXX | SUCCESS | rc=0 >>
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.xpg.com.br
 * epel: mirror.espoch.edu.ec
 * extras: centos.xpg.com.br
 * updates: centos.xpg.com.br
Package firewalld-0.4.3.2-8.1.el7_3.3.noarch already installed and latest version
Package glusterfs-fuse-3.7.9-12.el7.centos.x86_64 already installed and latest version
Nothing to do
Shared connection to app-dc-prd-osetcd03.XXX closed.


app-dc-prd-osetcd02.XXX | SUCCESS | rc=0 >>
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.xpg.com.br
 * epel: mirror.espoch.edu.ec
 * extras: centos.xpg.com.br
 * updates: centos.xpg.com.br
Package firewalld-0.4.3.2-8.1.el7_3.3.noarch already installed and latest version
Package glusterfs-fuse-3.7.9-12.el7.centos.x86_64 already installed and latest version
Nothing to do
Shared connection to app-dc-prd-osetcd02.XXX closed.


app-dc-prd-osmaster01.XXX | SUCCESS | rc=0 >>
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.netglobalis.net
 * epel: mirror.espoch.edu.ec
 * extras: mirror.netglobalis.net
 * updates: mirror.nbtelecom.com.br
Package firewalld-0.4.3.2-8.1.el7_3.3.noarch already installed and latest version
Package glusterfs-fuse-3.7.9-12.el7.centos.x86_64 already installed and latest version
Nothing to do
Shared connection to app-dc-prd-osmaster01.XXX closed.`

My GlusterFS cluster is standalone, I'm already using it for other applications and it's running just fine.

@SweetOps did you modify anything regarding iptables/firewalld? or is it just installed?

**Comment by jarrpa** [comment]
@jorgefromhell You need to open several ports. Please refer [here](https://github.com/gluster/gluster-kubernetes/blob/master/docs/setup-guide.md) for more details.

**Comment by SweetOps** [comment]
@jarrpa , small update: I've luck only with `RHEL 7 ` :(  on all other OS's same issue. Have you luck with the reproducing?

@jorgefromhell , on each CentOS\RHEL 7 server I've installed additional packages:

```sh
sudo yum install atomic  NetworkManager -y
```

and also additional variables in inventory:

```yaml
[OSEv3:vars]
os_firewall_use_firewalld=true

[glusterfs:vars]
r_openshift_storage_glusterfs_use_firewalld=true
r_openshift_storage_glusterfs_firewall_enabled=true
openshift_storage_glusterfs_timeout=600
openshift_storage_glusterfs_wipe=true
```

**Comment by jarrpa** [comment]
@SweetOps Does that include CentOS 7, or specifically RHEL?

**Comment by SweetOps** [comment]
@jarrpa What do you mean? If you mean inventory and packages - yes, for CentOS and RHEL

**Comment by SweetOps** [comment]
@jarrpa Could you write more about you environment (`OS`, version of `glusterfs` packages on hosts, `inventory` ) ?

**Comment by jarrpa** [comment]
@SweetOps My environment is non-Atomic CentOS/RHEL 7, using the [latest gluster/gluster-centos image](https://hub.docker.com/r/gluster/gluster-centos/), with the following inventory:
```
[OSEv3:children]
nodes
glusterfs
masters

[OSEv3:vars]
ansible_ssh_user=root
openshift_master_default_subdomain=cloudapps.example.com
deployment_type=origin
openshift_hosted_registry_storage_kind=glusterfs
openshift_disable_check=disk_availability,memory_availability,docker_storage
openshift_clock_enabled=False

[etcd]
master  openshift_public_ip=192.168.10.90 openshift_ip=192.168.10.90 openshift_public_hostname=master openshift_hostname=master node=True connect_to=192.168.10.90 storage=True master=True

[nfs]
master  openshift_public_ip=192.168.10.90 openshift_ip=192.168.10.90 openshift_public_hostname=master openshift_hostname=master node=True connect_to=192.168.10.90 storage=True master=True

[nodes]
master  openshift_public_ip=192.168.10.90 openshift_ip=192.168.10.90 openshift_public_hostname=master openshift_hostname=master node=True connect_to=192.168.10.90 storage=True master=True openshift_schedulable=False
node0  openshift_public_ip=192.168.10.100 openshift_ip=192.168.10.100 openshift_public_hostname=node0 openshift_hostname=node0 node=True connect_to=192.168.10.100 openshift_node_labels="{'region': 'infra'}" openshift_schedulable=True
node1  openshift_public_ip=192.168.10.101 openshift_ip=192.168.10.101 openshift_public_hostname=node1 openshift_hostname=node1 node=True connect_to=192.168.10.101 openshift_node_labels="{'region': 'infra'}" openshift_schedulable=True
node2  openshift_public_ip=192.168.10.102 openshift_ip=192.168.10.102 openshift_public_hostname=node2 openshift_hostname=node2 node=True connect_to=192.168.10.102 openshift_node_labels="{'region': 'infra'}" openshift_schedulable=True

[glusterfs]
node0 glusterfs_devices='[ "/dev/vdb", "/dev/vdc", "/dev/vdd" ]' glusterfs_ip=192.168.20.100
node1 glusterfs_devices='[ "/dev/vdb", "/dev/vdc", "/dev/vdd" ]' glusterfs_ip=192.168.20.101
node2 glusterfs_devices='[ "/dev/vdb", "/dev/vdc", "/dev/vdd" ]' glusterfs_ip=192.168.20.102

[masters]
master  openshift_public_ip=192.168.10.90 openshift_ip=192.168.10.90 openshift_public_hostname=master openshift_hostname=master node=True connect_to=192.168.10.90 storage=True master=True
```

**Comment by jorgefromhell** [comment]
uninstalled and deployed a couple more times and still not working. For me what's happening is that the pvc, even the dynamic endpoint gets created just fine, but when the deploy starts and the pv has to be mounted, I'm seeing this on the logs from the assigned node (endlessly, like 30-40 times per second, same message):

```
Sep  6 15:49:31 app-dc-prd-osnode03 journal: I0906 15:49:31.454781   23492 glusterfs.go:148] glusterfs: endpoints &Endpoints{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:glusterfs-dynamic-mariadb,GenerateName:,Namespace:teste,SelfLink:/api/v1/namespaces/teste/endpoints/glusterfs-dynamic-mariadb,UID:c32c7184-9333-11e7-8b7d-00505680c54c,ResourceVersion:15135,Generation:0,CreationTimestamp:2017-09-06 15:46:58 -0300 -03,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{gluster.kubernetes.io/provisioned-for-pvc: mariadb,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,},Subsets:[{[{10.20.0.50  <nil> nil} {10.20.0.51  <nil> nil} {10.20.0.52  <nil> nil}] [] [{ 1 TCP}]}],}
```

Also, @jarrpa , I got the names and types of installation mixed up, I'm using here non-Atomic CentOS 7 hosts but in my inventory I'm specifying the containerized installation. What should I be doing? Is containerized the best way to go? What matters to me really is that I can later upgrade my cluster, add nodes and do all the maintenance tasks it needs.

Thanks!

**Comment by jarrpa** [comment]
@jorgefromhell I have no experience using `containerized=true`, but I thought that was only if you were running on Atomic. @sdodson, can you clarify what it does and when it's used?

**Comment by jarrpa** [comment]
@SweetOps I just noticed that you list your GlusterFS variables in a `glusterfs:vars` section, which I don't officially support. I've had other users run into problems doing this that resolved when they brought them into the general `OSEv3:vars` section. Not saying that this is happening here, just FYI. :)

**Comment by jarrpa** [comment]
Actually, this is getting absurdly long.

@jorgefromhell, let's continue discussing your problem in #5118.

@SweetOps, if you feel your original issue has been resolved please close this one and open a new Issue with whatever problems you're still finding. :)

**Comment by SweetOps** [comment]
>I just noticed that you list your GlusterFS variables in a glusterfs:vars section, which I don't officially support.
@jarrpa , doesn't matter where that variables placed 

>if you feel your original issue has been resolved please close this one and open a new Issue with whatever problems you're still finding. :)

@jarrpa , sorry, but have luck only with `RHEL 7.3` with all other OS (`Centos` or any `Atomic`) problem (volumes doesn't mount to pods) is exist, so I don't issue is resolved.

**Comment by jarrpa** [comment]
@SweetOps Huh... are the errors you're getting in CentOS and the Atomic flavors the same? If so, could I ask you to either paste the errors again or link to a comment in this thread where you pasted them? I'm having a difficult time keeping track of all the data, sorry. :(

**Comment by SweetOps** [comment]
@jarrpa, Yes, the errors are same.

```sh
Events:
  FirstSeen	LastSeen	Count	From							SubObjectPath	Type		Reason		Message
  ---------	--------	-----	----							-------------	--------	------		-------
  8m		8m		1	default-scheduler							Normal		Scheduled	Successfully assigned jenkins-3-gwpwn to ip-10-0-1-165.eu-central-1.compute.internal
  6m		2m		3	kubelet, ip-10-0-1-165.eu-central-1.compute.internal			Warning		FailedMount	Unable to mount volumes for pod "jenkins-3-gwpwn_test(340d15e1-9352-11e7-a73f-0674f55ddaa5)": timeout expired waiting for volumes to attach/mount for pod "test"/"jenkins-3-gwpwn". list of unattached/unmounted volumes=[jenkins-data]
  6m		2m		3	kubelet, ip-10-0-1-165.eu-central-1.compute.internal			Warning		FailedSync	Error syncing pod
  8m		24s		12	kubelet, ip-10-0-1-165.eu-central-1.compute.internal			Warning		FailedMount	MountVolume.SetUp failed for volume "kubernetes.io/glusterfs/340d15e1-9352-11e7-a73f-0674f55ddaa5-pvc-885c65d6-9350-11e7-a73f-0674f55ddaa5" (spec.Name: "pvc-885c65d6-9350-11e7-a73f-0674f55ddaa5") pod "340d15e1-9352-11e7-a73f-0674f55ddaa5" (UID: "340d15e1-9352-11e7-a73f-0674f55ddaa5") with: glusterfs: mount failed: exit status 1 the following error information was pulled from the glusterfs log to help diagnose this issue: glusterfs: could not open log file for pod: jenkins-3-gwpwn
```

**Comment by jarrpa** [comment]
So this leaves us to ask what is different between your RHEL installations and your CentOS/Atomic installations... can you verify the following:
```
Each of the nodes that will host GlusterFS must also have appropriate firewall
rules for the required GlusterFS ports:
 * 2222  - sshd (if running GlusterFS in a pod)
 * 24007 - GlusterFS Management
 * 24008 - GlusterFS RDMA
 * 49152 to 49251 - Each brick for every volume on the host requires its own
   port. For every new brick, one new port will be used starting at 49152. We
   recommend a default range of 49152-49251 on each host, though you can adjust
   this to fit your needs.
The following kernel modules must be loaded:
 * dm_snapshot
 * dm_mirror
 * dm_thin_pool
For systems with SELinux, the following settings need to be considered:
 * virt_sandbox_use_fusefs should be enabled on each node to allow writing to
   remote GlusterFS volumes
In addition, for an OpenShift deployment you must:
 * Have 'cluster_admin' role on the administrative account doing the deployment
 * Add the 'default' and 'router' Service Accounts to the 'privileged' SCC
 * Have a router deployed that is configured to allow apps to access services
   running in the cluster
```

**Comment by TimoL** [comment]
I'm facing the exact same issue as @SweetOps.

Env:
- OpenShift Origin 3.6, containerized install
- CentOS Linux release 7.3.1611 (Core)
- OpenShift-Ansible (latest release-3.6 branch)
- Inventory: basically the same as @SweetOps 

@jarrpa I have verified the glusterfs node setup according to your checklist.

**firewall rules**: ok
```
glusterfs-1 $ sudo firewall-cmd --list-ports
10250/tcp 80/tcp 443/tcp 4789/udp 2222/tcp 24007/tcp 24008/tcp 49152-49251/tcp
```

**kernel modules**: not ok (`dm_snapshot` and `dm_mirror` missing)
```
glusterfs-1 $ cat /proc/modules | grep "dm_"
dm_thin_pool 65565 13 - Live 0xffffffffa0510000
dm_persistent_data 67216 1 dm_thin_pool, Live 0xffffffffa04f7000
dm_bio_prison 15907 1 dm_thin_pool, Live 0xffffffffa04e6000
dm_bufio 27972 1 dm_persistent_data, Live 0xffffffffa04de000
dm_mod 114430 55 dm_thin_pool,dm_persistent_data,dm_bufio, Live 0xffffffffa04c1000
libcrc32c 12644 3 openvswitch,dm_persistent_data,xfs, Live 0xffffffffa0096000
```

**selinux**: ok
```
glusterfs-1 $ getsebool virt_sandbox_use_fusefs
virt_sandbox_use_fusefs --> on
```

**OpenShift deployment**: ok

- `oadm policy add-role-to-user cluster-admin <username>` for deploying user was configured.
- service accounts in privileged SCC
  ```
  master-1 $ oc get scc privileged -o yaml
  [...]
  users:
  [...]
  - system:serviceaccount:glusterfs:default
  - system:serviceaccount:glusterfs:router
  [...]
  ```
- router is working fine

Thanks for your support!
Timo

**Comment by jarrpa** [comment]
@TimoL Did adjusting the kernel modules fix it?

**Comment by sdodson** [comment]
> @jorgefromhell I have no experience using containerized=true, but I thought that was only if you were running on Atomic. @sdodson, can you clarify what it does and when it's used?

Sorry I missed this question. You can install containerized on normal RHEL/Centos/Fedora but it defaults to RPM based installs so you need to set `containerized=true` to get a containerized install on non atomic hosts. On atomic, containerized is the only possibility so we automatically do it there.

**Comment by TimoL** [comment]
@jarrpa Got sidetracked by high-prio issue, will try asap and report back.

**Comment by besser82** [comment]
@jarrpa 
> Could you verify that the glusterfs-fuse package is installed? Could someone also quickly replicate the problem and then inspect the logs in /var/log/glusterfs on the OCP nodes running the GlusterFS pods to see if there's anything useful?

I did a fresh installation (Fedora 26 Atomic, containerized, registry storage on gluster).  The created GlusterFS volume(s) for the registry are mounted properly; GlusterFS volumes from custom created projects do not…  So gluster-fuse must be properly present on the nodes running the GlusterFS pods.

There must be some other problem going on…

**Comment by jarrpa** [comment]
@besser82 So if I understand you correctly, the installation successfully created the GlusterFS volume for the hosted registry, and that volume's PVC is mounted just fine by the registry pods. Is that right?

What do you mean by "GlusterFS volumes from custom-created projects"? Are you talking about trying to dynamically provision GlusterFS volumes? If so, you're still running into the `could not open log file` error?

**Comment by jarrpa** [comment]
So a statically provisioned GlusterFS volume is mounting into a pod successfully, but dynamically provisioned volumes are not...

Can you manually mount the dynamically provisioned GlusterFS volumes onto the host? Do you have all three of the required kernel modules installed?

**Comment by besser82** [comment]
@jarrpa I can manually mount all volumes without problems on all hosts…  Kernel modules are all present.

**Comment by jarrpa** [comment]
@besser82 So you're still getting something like the following error in the pod events:
`MountVolume.SetUp failed for volume "kubernetes.io/glusterfs/f297b47a-88c9-11e7-9976-0ab88d9eb221-pvc-d2362fbb-88c9-11e7-a6ad-0a862c7c70b3" (spec.Name: "pvc-d2362fbb-88c9-11e7-a6ad-0a862c7c70b3") pod "f297b47a-88c9-11e7-9976-0ab88d9eb221" (UID: "f297b47a-88c9-11e7-9976-0ab88d9eb221") with: glusterfs: mount failed: exit status 1 the following error information was pulled from the glusterfs log to help diagnose this issue: glusterfs: could not open log file for pod: gitlab-106655731-ndlm4`

Are you also seeing the errors mentioned in #5118?

Also, can you statically provision a GlusterFS volume, attach it to a PV/PVC, then try and mount that into a pod? If you need pointers, check these spots:
 * https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_storage_glusterfs/tasks/glusterfs_registry.yml#L81
 * https://github.com/openshift/openshift-ansible/blob/master/playbooks/common/openshift-glusterfs/registry.yml#L13-L25

**Comment by besser82** [comment]
@jarrpa I'm getting these errors in the events of the pod, as well as the messages in journal from #5118.

Statically provisioned volumes are working fine tho…

**Comment by jarrpa** [comment]
@besser82 So you've statically provisioned additional GlusterFS volumes and they mount to pods just fine??

I'm so baffled... can you confirm that dynamically provisioned volumes mount to pods when you deploy openshift with "containerized" not True?

**Comment by besser82** [comment]
@jarrpa Even that I can confirm.  If I'm installing 'rpm-style' all stuff works fine.

**Comment by besser82** [comment]
@jarrpa Thank you very much  =)

**Comment by jarrpa** [comment]
@besser82 Ah! One more thing: can you paste the exact commands and YAML you're running for both dynamic and static provisioning? That will help debug this much faster, as I'm currently making a number of assumptions. Include the YAML for things like endpoints too.

**Comment by besser82** [comment]
@jarrpa Will do so during tomorrow

**Comment by jarrpa** [comment]
@SweetOps @TimoL @besser82 I have a potential workaround. After you dynamically provision a GlusterFS volume, can you try deleting the dynamically created endpoints then manually creating an endpoints with the IP addresses of your GlusterFS nodes and see if the pods will come up?

**Comment by besser82** [comment]
@jarrpa Thank you, that works, tho it isn't a feasable solution in all use-cases…

**Comment by jarrpa** [comment]
@besser82 Certainly not! But unblocking at least one user is better than holding everyone up for a proper solution. :) Also this helps me in figuring out the actual symptom. Now if only I could find the root cause...

**Comment by jarrpa** [comment]
@besser82 Actually, if you're up for some more investigation, could you see if you can still reproduce the problem, capture the `oc get -o yaml` output of the "bad" Endpoints, then capture the same output of the "good" Endpoints after you recreate it?

**Comment by TimoL** [comment]
With a new OpenShift deployment I can not reproduce the issue any more. Currently it just works.
The new setup has seen tweaks in configuration, but I'm still using containerized install on CentOS 7.

Here's my diff of the old and new setup:

- CentOS was updated from 7.3.1611 to 7.4.1708
- now using multi-master with loadbalancer (single master before)
- iptables package was preinstalled on hosts, not installed by OpenShift-Ansible (had issue sometimes)
- `openshift_master_default_subdomain`, `openshift_master_cluster_hostname` and `openshift_master_cluster_public_hostname` were updated to new environment

**Comment by jarrpa** [comment]
As the OP has not responded in some time, I am closing this issue. If anyone is experiencing this exact or a similar problem, please open a new Issue.
